lua_modules/mcp23017/mcp23017.lua

```diff
@@ -0,0 +1,289 @@
+--[[
+
+	This Lua module provides access to the MCP23017 module.
+
+    The MCP23017 is a port expander and provides 16 channels for inputs and outputs.
+    Up to 8 devices (128 channels) are possible by the configurable address (A0 - A2 Pins).
+
+    Due to the 16 channels, 2 bytes are required for switching outputs or reading input signals. These are A and B.
+    A single pin can be set or a whole byte. The numbering of the individual pins starts at 0 and ends with 15.
+    The numbers up to 7 are on register GPIO A and from 8 to 15 on GPIO B.
+
+	The module requires `i2c` and `bit` C module built into firmware.
+
+
+	@author Marcel P. | Plomi.net
+	@github https://github.com/plomi-net
+
+	@version 1.0.0
+
+]]
+
+local i2c, string = i2c, string
+local issetBit = bit.isset
+local setBit = bit.set
+local clearBit = bit.clear
+
+-- MCP23017_ADDRESS = 0x20 -- (0x20 to 0x27)
+-- MCP23017_SDA = 2 -- D2/GPIO 4
+-- MCP23017_SCL = 1 -- D1/GPIO 5
+
+
+-- metatable
+local mcp23017 = {}
+mcp23017.__index = mcp23017
+
+-- registers (not used registers are commented out)
+local MCP23017_IODIRA = 0x00
+local MCP23017_IODIRB = 0x01
+local MCP23017_DEFVALA = 0x06
+local MCP23017_DEFVALB = 0x07
+local MCP23017_GPIOA = 0x12
+local MCP23017_GPIOB = 0x13
+
+--[[
+local MCP23017_IPOLA = 0x02
+local MCP23017_IPOLB = 0x03
+local MCP23017_GPINTENA = 0x04
+local MCP23017_GPINTENB = 0x05
+local MCP23017_DEFVALA = 0x06
+local MCP23017_DEFVALB = 0x07
+local MCP23017_INTCONA = 0x08
+local MCP23017_INTCONB = 0x09
+local MCP23017_IOCON = 0x0A
+local MCP23017_IOCON2 = 0x0B
+local MCP23017_GPPUA = 0x0C
+local MCP23017_GPPUB = 0x0D
+local MCP23017_INTFA = 0x0E
+local MCP23017_INTFB = 0x0F
+
+local MCP23017_INTCAPA = 0x10
+local MCP23017_INTCAPB = 0x11
+local MCP23017_OLATA = 0x14
+local MCP23017_OLATB = 0x15
+]]
+
+-- check device is available on address
+function mcp23017:checkDevice()
+    i2c.start(self.i2cId)
+    local response = i2c.address(self.i2cId, self.address, i2c.TRANSMITTER)
+    i2c.stop(self.i2cId)
+    if response ~= true then
+        print("MCP23017 device on " .. string.format('0x%02X', self.address) .. " not found")
+    end
+    return response
+end
+
+-- check device address (0x20 to 0x27)
+function mcp23017:checkAddress()
+    local addr = tonumber(self.address)
+    if (addr > 31 and addr < 40) then
+        return true
+    else
+        print("MCP23017 address is out of range")
+        return false
+    end
+end
+
+-- write byte
+function mcp23017:writeByte(registerAddr, val)
```

Comment:
Similarly to functions above - those are intended to be internal functions so they shouldn't be exported. The declaration could be rewritten to:
```lua
local function writeByte(self, registerAddr, val)
```
same goes to `readByte`, `getGPIORegisterAddr`, `checkPinIsInRange`, `writeToRegister`, `readFromRegister`

Reply:
Thanks also for this advice. This help me a lot.
I have now reprogrammed the internal function to internal.